### PR TITLE
Redefine of __is_hetero_execution_policy trait through __is_device_execution_policy and __is_fpga_execution_policy

### PR DESCRIPTION
**Pull request by SergeyKopienko**
The goal of this PR - to avoid extra code in policy traits implementations and redefine `__is_hetero_execution_policy` trait as [disjunction](https://en.cppreference.com/w/cpp/types/disjunction) of `__is_device_execution_policy` and `__is_device_execution_policy` traits:
```C++
template <typename _T>
using __is_hetero_execution_policy =
    ::std::disjunction<
      typename __is_device_execution_policy<_T>::type,
      typename __is_fpga_execution_policy<_T>::type>;
```

Previously we had more complicated implementation of `__is_hetero_execution_policy` trait:
```C++
template <typename _T>
struct __is_hetero_execution_policy : ::std::false_type
{
};

template <typename... PolicyParams>
struct __is_hetero_execution_policy<execution::device_policy<PolicyParams...>> : ::std::true_type
{
};

template <unsigned int unroll_factor, typename... PolicyParams>
struct __is_hetero_execution_policy<execution::fpga_policy<unroll_factor, PolicyParams...>> : ::std::true_type
{
};
```

But taking a look at the implementation of `__is_device_execution_policy` and `__is_fpga_execution_policy` :
```C++
template <typename _T>
struct __is_device_execution_policy : ::std::false_type
{
};

template <typename... PolicyParams>
struct __is_device_execution_policy<execution::device_policy<PolicyParams...>> : ::std::true_type
{
};
```
```C++
template <typename _T>
struct __is_fpga_execution_policy : ::std::false_type
{
};

template <unsigned int unroll_factor, typename... PolicyParams>
struct __is_fpga_execution_policy<execution::fpga_policy<unroll_factor, PolicyParams...>> : ::std::true_type
{
};
```
we are able to reimplement `__is_hetero_execution_policy` as proposed.

### CODE REVIEW

File: include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h
Comment (danhoeflinger) [inline include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h:261]: I think, `Args` should instead be `PolicyParams` to match the others here.

Comment (SergeyKopienko) [reply include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h]: Fixed

**Review by danhoeflinger — DISMISSED** [review]
LGTM, this also covers half of  #1298.

Please wait until CI has run and is green to merge.

File: include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h
Comment (SergeyKopienko) [inline include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h:234]: This variadic template param has been changed to `typename _T` because variadic is not required here.

File: include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h
Comment (SergeyKopienko) [inline include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h:235]: This variadic template param has been changed to `typename _T` because variadic is not required here.

**Review by danhoeflinger — APPROVED** [review]
LGTM, now this completely covers #1298, and we can close that when this is merged.

**Pull request merged**
